__init__.py

```diff
@@ -101,6 +104,78 @@ def handle_median(self, message):
         except IndexError:
             self.speak_dialog('IndexError', {'func': func})
 
+    def cluster_validator(self, response):
+        requested_adjustments = []
+        for adjustment in self.cluster_adjustments:
+            if adjustment in response:
+                requested_adjustments.append(adjustment)
+        return requested_adjustments
+
+    @intent_file_handler('cluster.intent')
+    def handle_cluster(self, message):
+
+        func = "clusteranalysis"
+        filename = message.data.get('file')
+        x_col = message.data.get('colname_x').lower()
+        y_col = message.data.get('colname_y').lower()
+        num_clusters = w2n.word_to_num(message.data.get('num_clusters'))
+
+        title = None
+        x_label = None
+        y_label = None
+
+        try:
+            file_handler = FileHandler(filename)
```

Comment:
this should be in own function, because we have this in every intent_handle function